hnn_core/viz.py

```diff
@@ -425,12 +447,9 @@ def plot_spikes_raster(cell_response, ax=None, show=True):
         if cell_type_times:
             cell_type_times = np.concatenate(cell_type_times)
             cell_type_ypos = np.concatenate(cell_type_ypos)
-        else:
-            cell_type_times = []
-            cell_type_ypos = []
 
-        ax.scatter(cell_type_times, cell_type_ypos, label=cell_type,
-                   color=cell_type_colors[cell_type])
+            ax.scatter(cell_type_times, cell_type_ypos, label=cell_type,
```

Comment:
This prevents cells that didn't fire in the simulation from being included in the legend